splash/render_options.py

```diff
@@ -170,14 +193,20 @@ def get_headers(self):
 
         if isinstance(headers, (list, tuple)):
             for el in headers:
-                if not (isinstance(el, (list, tuple)) and len(el) == 2 and all(isinstance(e, basestring) for e in el)):
+                if not (isinstance(el, (list, tuple)) and len(el) == 2 and all(
+                        isinstance(e, (six.string_types, six.binary_type)) for e
+                        in el)):
                     raise BadOption("'headers' must be either JSON array of (name, value) pairs or JSON object")
-
+        try:
+            headers = bytes_to_unicode(headers, encoding='ascii')
+        except UnicodeDecodeError:
+            raise BadOption("headers can not contain non-ascii characters.")
```

Comment:
What if they must contain these characters, i.e. what if user wants to send non-ascii cookies? Many servers accept utf8-encoded cookies.